src/CommandRunner.cpp

```diff
@@ -249,23 +249,28 @@ Try<Nothing> runCommandWithTimeout(const std::string& command,
     return false;
   };
 
-  milliseconds tickPeriod(100);
-  Timer t1(tickPeriod, milliseconds(timeoutInSeconds * 1000));
-  t1.run(waitProcess);
-
-  if (t1.hasTimedOut() && !forceKillRequired) {
-    TASK_LOG(WARNING, loggingMetadata)
-        << "External command took too long to exit. "
-        << "Sending SIGTERM...";
-    if (kill(pid, SIGTERM) == -1) {
-      TASK_LOG(ERROR, loggingMetadata) << "Failed to send SIGTERM: "
-                                       << strerror(errno);
-      forceKillRequired = true;
-    } else {
-      Timer t(tickPeriod, milliseconds(1000));
-      t.run(waitProcess);
-      if (t.hasTimedOut()) {
+  // Hack to support command without timeout
+  // TODO: to remove once the CommandIsolator::watch implementation has been
+  // changed
+  if (timeoutInSeconds > 0) {
```

Comment:
Why do you need to skip the SIGTERM?

Reply:
would not it be easier to just set the timeout to 1 and avoid complexifying this part?

Reply:
I oversaw the termination handling part.
We can't set it to 1, the command has to be running for as long as the container is here. I'll think about something, or discard this part, knowing that an implementation change is on its way afterwards.